resources/qml/MonitorButton.qml

```diff
@@ -0,0 +1,183 @@
+        }
+        else if(Cura.MachineManager.printerOutputDevices[0].jobState == "pre_print")
+        {
+            return "Preparing..."
+        }
+        else
+        {
+            return " "
+        }
+
+    }
+
+    Label
+    {
+        id: statusLabel
+        width: parent.width - 2 * UM.Theme.getSize("default_margin").width
+        anchors.top: parent.top
+        anchors.left: parent.left
+        anchors.leftMargin: UM.Theme.getSize("default_margin").width
+
+        color: printerConnected ? Cura.MachineManager.printerOutputDevices[0].jobState == "paused" ? UM.Theme.getColor("status_paused") : UM.Theme.getColor("status_ready") : UM.Theme.getColor("status_offline")
+        font: UM.Theme.getFont("large")
+        text: statusText;
+    }
+
+    Label
+    {
+        id: percentageLabel
+        anchors.top: parent.top
+        anchors.right: progressBar.right
+
+        color: printerConnected ? Cura.MachineManager.printerOutputDevices[0].jobState == "paused" ? UM.Theme.getColor("status_paused") : UM.Theme.getColor("status_ready") : UM.Theme.getColor("status_offline")
+        font: UM.Theme.getFont("large")
+        text: Math.round(progress * 100) + "%";
+        visible: printerConnected
+    }
+
+    Rectangle
+    {
+        id: progressBar
+        width: parent.width - 2 * UM.Theme.getSize("default_margin").width
+        height: UM.Theme.getSize("progressbar").height
+        anchors.top: statusLabel.bottom
+        anchors.topMargin: UM.Theme.getSize("default_margin").height / 4
+        anchors.left: parent.left
+        anchors.leftMargin: UM.Theme.getSize("default_margin").width
+        radius: UM.Theme.getSize("progressbar_radius").width
+        color: UM.Theme.getColor("progressbar_background")
+
+        Rectangle
+        {
+            width: Math.max(parent.width * base.progress)
+            height: parent.height
+            color: printerConnected ? Cura.MachineManager.printerOutputDevices[0].jobState == "paused" ? UM.Theme.getColor("status_paused") : UM.Theme.getColor("status_ready") : UM.Theme.getColor("status_offline")
+            radius: UM.Theme.getSize("progressbar_radius").width
+        }
+    }
+
+    Button
+    {
+        id: abortButton
+
+        visible: printerConnected
+        height: UM.Theme.getSize("save_button_save_to_button").height
+
+        anchors.top: progressBar.bottom
+        anchors.topMargin: UM.Theme.getSize("default_margin").height
+        anchors.right: parent.right
+        anchors.rightMargin: UM.Theme.getSize("default_margin").width
+
+        text: catalog.i18nc("@label:", "Abort Print")
+        onClicked: { Cura.MachineManager.printerOutputDevices[0].setJobState("abort") }
+
+
+        style: ButtonStyle
+        {
+            background: Rectangle
+            {
+                border.width: UM.Theme.getSize("default_lining").width
+                border.color: !control.enabled ? UM.Theme.getColor("action_button_disabled_border") :
```

Comment:
I have been changing these in other places as well, please do not use so many nested ternary operators and instead just write them out as a proper if-then-else construction. This gets really hard to follow otherwise.